app/V1Module/presenters/UsersPresenter.php

```diff
@@ -430,4 +431,31 @@ public function actionExercises(string $id) {
     $this->sendSuccessResponse($user->getExercises()->getValues());
   }
 
+  public function checkSetRole(string $id) {
+    $user = $this->users->findOrThrow($id);
+    if (!$this->userAcl->canSetRole($user)) {
```

Comment:
Which users can set roles? A user with this privilege can make anybody a superadmin, which is pretty dangerous. Maybe we might want to have a non-admin user who can make supervisors...

Reply:
Like above, do you suggest that we should do the checks of this in the endpoint?